data/cognates/Collecting_cognates_dataset.ipynb

```diff
@@ -0,0 +1,489 @@
+    "\n",
+    "        children = []\n",
+    "        for i in range(len(curr_list) // 2):\n",
+    "            children.append([curr_list[2 * i].replace('\"', ''), curr_list[2 * i + 1]])\n",
+    "        return children\n",
+    "    return []"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Writing complete\n"
+     ]
+    }
+   ],
+   "source": [
+    "import csv\n",
+    "\n",
+    "MAXN = 10\n",
+    "\n",
+    "file = open('cognates_full.csv', 'w')\n",
+    "writer = csv.writer(file)\n",
+    "writer.writerows([['first_word', 'second_word']])\n",
+    "\n",
+    "with file:\n",
+    "    first_word = list()\n",
+    "    second_word = list()\n",
+    "    for key in alphabet:\n",
+    "        for i in range(alphabet[key][0]):\n",
+    "            curr_link = alphabet[key][1]\n",
+    "            if i != 0:\n",
+    "                curr_link += \"/pn\"\n",
+    "                curr_link += str(i + 1)\n",
+    "            word_list = get_children(curr_link)\n",
+    "            for elem in word_list:\n",
+    "                child_page = \"https://cognates.ru\" + elem[0]\n",
+    "                cognates = get_children(child_page)\n",
+    "                for i in range(min(MAXN, len(cognates))):\n",
+    "                    cognate = cognates[i]\n",
+    "                    first_word.append(elem[1])\n",
+    "                    second_word.append(cognate[1])\n",
+    "                    writer = csv.writer(file)\n",
+    "                    writer.writerows([[elem[1], cognate[1]]])\n",
+    "print(\"Writing complete\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/html": [
+       "<div>\n",
+       "<style scoped>\n",
+       "    .dataframe tbody tr th:only-of-type {\n",
+       "        vertical-align: middle;\n",
+       "    }\n",
+       "\n",
+       "    .dataframe tbody tr th {\n",
+       "        vertical-align: top;\n",
+       "    }\n",
+       "\n",
+       "    .dataframe thead th {\n",
+       "        text-align: right;\n",
+       "    }\n",
+       "</style>\n",
+       "<table border=\"1\" class=\"dataframe\">\n",
+       "  <thead>\n",
+       "    <tr style=\"text-align: right;\">\n",
+       "      <th></th>\n",
+       "      <th>first_word</th>\n",
+       "      <th>second_word</th>\n",
+       "    </tr>\n",
+       "  </thead>\n",
+       "  <tbody>\n",
+       "    <tr>\n",
+       "      <th>0</th>\n",
+       "      <td>А</td>\n",
+       "      <td>А-конто</td>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th>1</th>\n",
+       "      <td>А</td>\n",
+       "      <td>Аканье</td>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th>2</th>\n",
+       "      <td>А</td>\n",
+       "      <td>Акать</td>\n",
```

Comment:
Let's compute another metric on our dataset: longest common subsequence. Then, we can use it in absolute difference operation between word pairs. I think, this metric can be robust to alternating letters in word's root :)

More information here: https://ru.wikipedia.org/wiki/%D0%9D%D0%B0%D0%B8%D0%B1%D0%BE%D0%BB%D1%8C%D1%88%D0%B0%D1%8F_%D0%BE%D0%B1%D1%89%D0%B0%D1%8F_%D0%BF%D0%BE%D0%B4%D0%BF%D0%BE%D1%81%D0%BB%D0%B5%D0%B4%D0%BE%D0%B2%D0%B0%D1%82%D0%B5%D0%BB%D1%8C%D0%BD%D0%BE%D1%81%D1%82%D1%8C